File: SWP391.v1/EV-Service-BE/EVMaintenanceSystem/src/main/java/com/swp/evmsystem/service/SparePartService.java
package com.swp.evmsystem.service;

import com.swp.evmsystem.dto.request.SparePartRequestDTO;
import com.swp.evmsystem.dto.response.SparePartResponseDTO;
import com.swp.evmsystem.dto.response.SparePartStatsDTO;
import com.swp.evmsystem.entity.SparePartEntity;

import java.util.List;

public interface SparePartService {
    List<SparePartEntity> getAllSpareParts();
    
    List<SparePartEntity> getInStockParts();
    
    SparePartEntity getSparePartById(Integer id);
    
    SparePartEntity createSparePart(SparePartEntity sparePart);
    
    SparePartEntity updateSparePart(SparePartEntity sparePart);
    
    void deleteSparePart(Integer id);
    
    List<SparePartResponseDTO> getAllSparePartsDTO();
    
    SparePartResponseDTO createSparePartDTO(SparePartRequestDTO request);
    
    SparePartResponseDTO updateSparePartDTO(Integer id, SparePartRequestDTO request);
    
    void deleteSparePartById(Integer id);
    
    SparePartStatsDTO getStatistics();
}

File: EV-Service-FE/src/services/sparePartService.js
import api from './api';

export const sparePartService = {
  // Lấy tất cả spare parts
  getAllSpareParts: async () => {
    const response = await api.get('/spare-parts');
    return response;
  },
  
  // Lấy spare parts còn hàng
  getInStockParts: async () => {
    const response = await api.get('/spare-parts/in-stock');
    return response;
  },
  
  // Lấy spare part theo ID
  getSparePartById: async (id) => {
    const response = await api.get(`/spare-parts/${id}`);
    return response;
  },
  
  // Admin: Thêm spare part mới
  createSparePart: async (sparePartData) => {
    const response = await api.post('/spare-parts', sparePartData);
    return response;
  },
  
  // Admin: Cập nhật spare part
  updateSparePart: async (id, sparePartData) => {
    const response = await api.put(`/spare-parts/${id}`, sparePartData);
    return response;
  },
  
  // Admin: Xóa spare part
  deleteSparePart: async (id) => {
    const response = await api.delete(`/spare-parts/${id}`);
    return response;
  },
};
